index.js

```diff
@@ -1,88 +1,239 @@
-          group: 'internal',
-          position: 'after'
+/**
+ * @typedef {Parameters<typeof antfu>[1]} UserConfig
+ * @param {...UserConfig} userConfigs - User-defined ESLint configurations to extend or override
+ * @returns {ReturnType<typeof antfu>} A chainable ESLint config
+ */
+export default function globalbrain(...userConfigs) {
+  return (
+    antfu(
+      {
+        // `type: 'lib'` only adds [one rule](https://github.com/antfu/eslint-config/blob/0cd12cc90d2100798a5d8f5d51b34753b7be7f70/src/configs/typescript.ts#L165-L173),
+        // as our projects are mostly apps, we can just use the default app config,
+        // and either provide a separate config for lib projects,
+        // or manually enable the rule in the projects if needed
+
+        // TS/Vue features are auto-detected,
+        // but to avoid errors being thrown from our overrides,
+        // we need to explicitly enable them here.
+        typescript: true,
+        vue: true,
+
+        // We are more conservative on curly braces
+        // <https://github.com/antfu/eslint-config/blob/0cd12cc90d2100798a5d8f5d51b34753b7be7f70/src/configs/stylistic.ts#L54-L63>
+        lessOpinionated: true
+      },
+      {
+        rules: {
+          'max-statements-per-line': 'off'
         }
-      ],
-      alphabetize: {
-        order: 'asc',
-        caseInsensitive: false
       }
-    }],
-    'new-cap': ['error', {
-      newIsCap: false,
-      capIsNew: false,
-      properties: true
-    }],
-    'n/prefer-global/process': 'off',
+    )
+      .override('antfu/javascript/rules', {
+        rules: {
+          'no-cond-assign': 'off',
+          'new-cap': [
+            'error',
+            {
+              newIsCap: false,
+              capIsNew: false,
+              properties: true
+            }
+          ]
+        }
+      })
+      .override('antfu/stylistic/rules', {
+        rules: {
+          'style/arrow-parens': ['error', 'always'],
+          'style/brace-style': [
+            'error',
+            '1tbs',
+            {
+              allowSingleLine: true
+            }
+          ],
+          'style/comma-dangle': ['error', 'never']
+        }
+      })
+      .override('antfu/typescript/rules', {
+        rules: {
+          'ts/consistent-type-definitions': 'off',
+          'ts/consistent-type-imports': [
+            'error',
+            {
+              prefer: 'type-imports',
+              disallowTypeAnnotations: false,
+              fixStyle: 'inline-type-imports'
+            }
+          ],
+          'ts/no-empty-interface': 'off',
+          // Can't enable this rule because it conflicts with our preference of inline type imports.
+          'ts/no-import-type-side-effects': 'off',
+          'ts/no-unused-vars': [
+            'error',
+            {
+              argsIgnorePattern: '^_',
+              destructuredArrayIgnorePattern: '^_'
+            }
+          ],
+          'ts/no-use-before-define': 'off'
```

Comment:
(line 81) fine to enable it. it's enabled for variables only and seems correct to do so.

Reply:
Got 90+ violations in Deus client codebase and can't be autofixed. They are easily fixable with AI though.

Because many `ref`/`computed` are used in event handler definitions before they are defined.

Reply:
Ah, hmmm, yeah this tough one 🤔  I think we should at least enable this on variables? Because I think it is useful to define like private functions after the public one. Like, it kinda reads better.

```ts
function useX() {
  function publicMainMethod() {
    return _subMethod()
  }
  
  function _subMethod() {
    // ...
  }
  
  return {
    publicMainMethod
  }
}
```

Reply:
damn <img src="https://github.com/user-attachments/assets/3db19a67-7c86-4857-b0a9-1072634209f6" height="20">

leave it then, we can enable it later